design/src/rfcs/rfc0012_unsynchronized_versioning.md

```diff
@@ -0,0 +1,207 @@
+    - Model updates that _only_ have documentation changes
+
+During phase 3, bumps to the `major` version must be coordinated across all SDK and runtime crates.
+
+### Release Identification
+
+Since there will no longer be one SDK "version", release tags will be dates in `YYYY-MM-DD` format
+rather than version numbers. Additionally, the SDK's user agent string will need to include a separate
+service version number (this requirement has already been implemented).
+
+### Yanking
+
+It must be possible to yank an entire release with a single action. The publisher tool must
+be updated to understand which crate versions were released with a given release tag, and be able to
+yank all the crates published from that tag.
+
+Phase 1: Unsynchronized Dev Preview
+-----------------------------------
+
+Phase 1 will address the following challenges introduced by uncoordinating the `major.minor` versions:
+- Tracking of versions associated with a release tag
+- Creation of version bump process for code generated crates
+- Enforcement of version bump process in runtime crates
+- Yanking of versions associated with a release tag
+
+### Version Tracking
+
+A new manifest file will be introduced in the root of [aws-sdk-rust] named `versions.toml` that describes
+all versioning information for any given commit in the repository. In the main branch, the `versions.toml`
+in tagged commits will become the source of truth for which crate versions belong to that release, as well
+as additional metadata that's required for maintaining version process in the future.
+
+The special `0.0.0-smithy-rs-head` version that is used prior to Phase 1 for maintaining the runtime crate
+versions will no longer be used (as detailed in [Versioning for Runtime Crates](#versioning-for-runtime-crates)).
+
+This format will look as follows:
+```toml
+smithy_rs_version = "<release-tag|commit-hash>"
+
+[aws-smithy-types]
+version = "0.50.1"
+
+[aws-config]
+version = "0.40.0"
+
+[aws-sdk-s3]
+version = "0.89.0"
+model_hash = "<hash>"
+
+# ...
+```
+
+The auto-sync tool is responsible for maintaining this file. When it generates a new SDK, it will take
+the version numbers from runtime crates directly, and it will use the rules from the next section to determine
+the version numbers for the generated crates.
+
+### Versioning for Code Generated (SDK Service) Crates
+
+Code generated crates will have their `minor` version bumped when the version of [smithy-rs] used to generate
+them changes, or when model updates with API changes are made. Three pieces of information are required to
+handle this process: the previously released version number, the [smithy-rs] version used to generate the code,
+and the level of model updates being applied. For this last one, if there are multiple model updates that
+affect only documentation, but then one model update that affects an API, then as a whole they will be
+considered as affecting an API and require a `minor` version bump.
```

Comment:
_paraphrasing this to make sure I understand_
>"if we update 2+ models in a release, and at least 1 of those models has API changes, all updated models will receive a `minor` version bump"

Is that right?

Reply:
So imagine we have the following updates to the S3 model, and the S3 version was previously `1.2.3`:

1. Documentation
2. Documentation
3. API change
4. API change

The new version in this case, will be exactly one minor version bump: `1.3.0`